src/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -284,13 +283,12 @@ public byte[] ToByteArray()
             byte highByte;
             int nonZeroDwordIndex = 0;
             uint highDword;
-            uint[] bits = _bits;
-            if (bits == null)
+            if (_bits == null)
             {
-                highByte = (byte)((sign < 0) ? 0xff : 0x00);
-                highDword = unchecked((uint)sign);
+                highByte = (byte)((_sign < 0) ? 0xff : 0x00);
+                highDword = unchecked((uint)_sign);
```

Comment:
@dtrebbien, thank you very much for your help! I'll change the code back to using locals then.

(A bit disappointing, though; that the JITer doesn't include some magic to optimize that. I really thought using `readonly` leads to some kind of equivalent access. My knowledge of the CLR seems to need some improvements...)